k2/csrc/ragged_test.cu

```diff
@@ -2559,24 +2559,42 @@ TEST(RaggedShapeOpsTest, RaggedShapeAxis0Splitter) {
 }
 template <typename T>
 static void TestSegmentedExclusiveSum() {
-  std::vector<T> expected_v = {0, 1, 3, 6,
-                               //
-                               0, 3, 7,
-                               //
-                               0, 5, 11, 18};
-
   for (auto &c : {GetCpuContext(), GetCudaContext()}) {
-    Ragged<T> src("[ [1 2 3 -1] [3 4 -1] [] [5 6 7 -1] ]");
-    src = src.To(c);
-    Array1<T> dst(c, src.values.Dim());
-
-    SegmentedExclusiveSum(src, &dst);
-
-    Array1<T> expected(c, expected_v);
-    CheckArrayData(dst, expected);
-
-    SegmentedExclusiveSum(src, &src.values);
-    CheckArrayData(src.values, expected);
+    {
+      // simple case
+      Ragged<T> src("[ [1 2 3 -1] [3 4 -1] [] [5 6 7 -1] ]");
+      src = src.To(c);
+      Array1<T> dst(c, src.NumElements());
+      SegmentedExclusiveSum(src, &dst);
+
+      std::vector<T> expected = {0, 1, 3, 6,
+                                 //
+                                 0, 3, 7,
+                                 //
+                                 0, 5, 11, 18};
+      CheckArrayData(dst, expected);
+
+      // &src.values == dst
+      SegmentedExclusiveSum(src, &src.values);
+      CheckArrayData(src.values, expected);
+    }
+    {
+      // random case, we assume the implementation for cpu is correct and only
```

Comment:
For testing, I would suggest using ExclusiveSum to compute the ground truth for each sublist separately.

Reply:
mm, the underlying cpu are the same, so there's no difference. Anyway, we have tested cpu code with simple case, so I believe it's fine to just compare the result of GPU version and CPU version.